### ISSUE TITLE
refactor the click() function so that it throws an exception and returns if the parameter is not found

### ISSUE THREAD

**Issue by psilaghi** [mozilla/iris#375]
Refactor the click() function so that it throws an exception and returns if the parameter is not found

**Comment by mwobensmith** [comment]
This is causing problems in some test cases.

I see that this line 1142 in core.py is activated when the image is not found:
```

1139    if possible_offset is not None:
1140        _click_at(Location(p_top.x + possible_offset.x, p_top.y + possible_offset.y), clicks, duration, button)
1141    else:
1142        _click_at(Location(p_top.x + width / 2, p_top.y + height / 2), clicks, duration, button)
```

It also appears that the default wait time to search for the image isn't respected for the `click` method. 

On Mac, I am seeing the web compat test `bing.py` fail in line 47, because the pattern that `click` is looking for is not found immediately. This causes Iris to click on the Apple logo in the upper left corner, and the test itself performs many keyboard commands that create more problems.

Lastly, when I tried to debug this using `-i DEBUG`, I noticed that no debug image was saved to disk for the pattern passed to `click`. Not sure why that is, but we should have that, I think.

**Comment by mwobensmith** [comment]
This is causing the `settings.py` unit test to fail on my (very slow) Linux environment. The `click` is called before the YouTube image in the page has loaded, and it does not wait, so it raises an exception.
```

» Traceback (most recent call last):                                                                         «
»   File "/home/mozilla/iris/iris/test_runner.py", line 54, in run                                           «
»     current.run()                                                                                          «
»   File "/home/mozilla/iris/iris/tests/unit_tests/settings.py", line 70, in run                             «
»     click(youtube_top_site_image)                                                                          «
»   File "/home/mozilla/iris/iris/api/core.py", line 1530, in click                                          «
»     _general_click(where, 1, duration, in_region, 'left')                                                  «
»   File "/home/mozilla/iris/iris/api/core.py", line 1493, in _general_click                                 «
»     _click_pattern(pattern, clicks, duration, in_region, button)                                           «
»   File "/home/mozilla/iris/iris/api/core.py", line 1443, in _click_pattern                                 «
»     raise FindError('Unable to click on: %s' % pattern.image_path)                                         «
» FindError: Unable to click on: /home/mozilla/iris/iris/tests/experiments/images/linux/youtube_top_site.png
```

**Comment by mwobensmith** [comment]
Merged in #444
